code/modules/clothing/clothing.dm

```diff
@@ -198,6 +200,28 @@ BLIND     // can't see anything
 		var/mob/M = loc
 		M.update_inv_shoes()
 
+/obj/item/clothing/shoes/attackby(obj/item/I, mob/user, params)
+	if(!can_hold_knives)
+		return
+	if(held_knife)
+		user << "<span class='notice'>There's already something in [src].</span>"
+		return
+	if(istype(I, /obj/item/weapon/kitchen/knife) || istype(I, /obj/item/weapon/pen))//can hold both regular pens and energy daggers. made for your every-day tactical librarians/murderers.
+		if(!user.drop_item())
+			return
+		I.loc = src
+		user << "<span class='notice'>You discreetly slip [I] into [src]. Alt-click [src] to remove it.</span>"
+		held_knife = I
+
+/obj/item/clothing/shoes/AltClick(mob/user)
+	if(user.incapacitated() || !held_knife || !can_hold_knives)
+		return
+	if(!user.put_in_hands(held_knife))
+		user << "<span class='notice'>You fumble for [held_knife] and it falls on the floor.</span>"
+		return 1
+	user.visible_message("<span class='warning'>[user] draws [held_knife] from their shoes!</span>", "<span class='notice'>You withdraw [held_knife] from [src].</span>")
+	held_knife = null
+	add_fingerprint(user)

```

Comment:
actually i dont think this line is needed at all on second thought, you get fingerprints on the knife from holding it already and to put shoes on your feet covers them in fingerprints to begin with. i'll just delete this line